src/analysis/normed_space/dual.lean

```diff
@@ -96,3 +117,160 @@ end
 end bidual_isometry
 
 end normed_space
+
+namespace inner_product_space
+open is_R_or_C continuous_linear_map
+
+variables (𝕜 : Type*)
+variables {E : Type*} [is_R_or_C 𝕜] [inner_product_space 𝕜 E]
+local notation `⟪`x`, `y`⟫` := @inner 𝕜 E _ x y
+local postfix `†`:90 := @is_R_or_C.conj 𝕜 _
+
+/--
+Given some `x` in an inner product space, we can define its dual as the continuous linear map
+`λ y, ⟪x, y⟫`.
+-/
+def to_dual' (x : E) : normed_space.dual 𝕜 E :=
+linear_map.mk_continuous
+  { to_fun := λ y, ⟪x, y⟫,
+    map_add' := λ _ _, inner_add_right,
+    map_smul' := λ _ _, inner_smul_right }
+  ∥x∥
+  (λ y, by { rw [is_R_or_C.norm_eq_abs], exact abs_inner_le_norm _ _ })
+
+@[simp] lemma to_dual'_apply {x y : E} : to_dual' 𝕜 x y = ⟪x, y⟫ := rfl
+
+/-- In an inner product space, the norm of the dual of a vector `x` is `∥x∥` -/
+@[simp] lemma to_dual'_isometry (x : E) : ∥to_dual' 𝕜 x∥ = ∥x∥ :=
+begin
+  refine le_antisymm _ _,
+  { exact linear_map.mk_continuous_norm_le _ (norm_nonneg _) _ },
+  { cases eq_or_lt_of_le (norm_nonneg x) with h h,
+    { have : x = 0 := norm_eq_zero.mp (eq.symm h),
+      simp [this] },
+    { refine (mul_le_mul_right h).mp _,
+      calc ∥x∥ * ∥x∥ = ∥x∥ ^ 2 : by ring
+      ... = re ⟪x, x⟫ : norm_sq_eq_inner _
+      ... ≤ abs ⟪x, x⟫ : re_le_abs _
+      ... = ∥to_dual' 𝕜 x x∥ : by simp [norm_eq_abs]
+      ... ≤ ∥to_dual' 𝕜 x∥ * ∥x∥ : le_op_norm (to_dual' 𝕜 x) x } }
+end
+
+variables {F : Type*} [inner_product_space ℝ F]
+
+/-- In an inner product space `F`, the function that takes a vector `x` in `F` to its dual
+`λ y, ⟪x, y⟫` is a continuous linear map. -/
+-- TODO extend to `is_R_or_C` (requires a definition of conjugate linear maps)
+def to_dual_map : F →L[ℝ] (normed_space.dual ℝ F) :=
```

Comment:
I guess you can define an `ℝ`-linear map in the `is_R_or_C` case too? And add a note that in the complex case it is actually a conjugate linear map but we don't have them in mathlib.

BTW, the docstrings of all non-canonical forms (`to_dual'`, `to_dual_map`) should say that users should use `to_dual` instead.

Reply:
`to_dual_map` is not a non-canonical form, because it is the best thing available when the space is not complete.